scripts/tools/initialize_virtualenv.py

```diff
@@ -0,0 +1,136 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+if __name__ not in ("__main__", "__mp_main__"):
+    raise SystemExit(
+        "This file is intended to be executed as an executable program. You cannot use it as a module."
+        f"To run this script, run the ./{__file__} command"
+    )
+
+
+def clean_up_airflow_home(airflow_home: Path):
+    if airflow_home.exists():
+        print(f"Removing {airflow_home}")
+        shutil.rmtree(airflow_home, ignore_errors=True)
+
+
+def check_if_in_virtualenv() -> bool:
+    return hasattr(sys, 'real_prefix') or (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix)
+
+
+def pip_install_requirements() -> int:
+    """
+    install the requirements of the current python version.
+    return 0 if success, anything else is an error.
+    """
+    version = get_python_version()
+    constraint = (
+        f"https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-{version}.txt"
+    )
+    pip_install_command = ["pip", "install", "-e", ".[devel-all]", "--constraint", constraint]
```

Comment:
Actuallly it woudl be great if you can make the "extra" configurable. The "devel_all" extra might be not installable in all circumstances (some libraries might be missing). So while it should be the default, we should be able to give the user opportunity to run it with a different extra or set of those.

So there should be an extra parameter that you can add

./scripts/tools/initialize_virtualenv.py "devel"

or 

./scripts/tools/initialize_virtualenv.py "devel,google"

That should enable the user to initialize the venv with other extras.

And whenever we have an error while installing, we shoudl inform the user about it. See the other comment: